In [1]:
import time
import numpy as np
import pandas as pd

In [2]:
# load 2000 data
bush = pd.read_csv( '../data/2000/age-groups-and-sex-census-DEC_00_SF1_QTP1/DEC_00_SF1_QTP1_with_ann.csv' , low_memory=False )
# load 2010 data
obama = pd.read_csv( '../data/2010/age-groups-and-sex-census-DEC_10_SF1_QTP1/DEC_10_SF1_QTP1_with_ann.csv' , low_memory=False )
# load 2000 metadata
b_metadata = pd.read_csv( '../data/2000/age-groups-and-sex-census-DEC_00_SF1_QTP1/DEC_00_SF1_QTP1_metadata.csv' )
# load 2010 metadata
o_metadata = pd.read_csv( '../data/2010/age-groups-and-sex-census-DEC_10_SF1_QTP1/DEC_10_SF1_QTP1_metadata.csv' )

In [3]:
# same metadata , should make for good comps
b_metadata.info() == o_metadata.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 289 entries, 0 to 288
Data columns (total 2 columns):
GEO.id    289 non-null object
Id        289 non-null object
dtypes: object(2)
memory usage: 4.6+ KB
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 289 entries, 0 to 288
Data columns (total 2 columns):
GEO.id    289 non-null object
Id        289 non-null object
dtypes: object(2)
memory usage: 4.6+ KB


True

In [4]:
bush.describe()

GEO.id GEO.id2 GEO.display-label HC02_VC01 HC03_VC01  \
count            33179   33179             33179     33179     33179   
unique           33179   33179             33179     14832     11189   
top     8600000US21703   72683       ZCTA5 66526         0         0   
freq                 1       1                 1       935       942   

       HC04_VC01 HC06_VC01 HC07_VC01 HC08_VC01 HC09_VC01  ... HC07_VC40  \
count      33179     33179     33179     33179     33179  ...     33179   
unique     11348         3         3         3      1239  ...       282   
top            0     100.0     100.0     100.0       (X)  ...       (X)   
freq         946     32243     32236     32232       946  ...       942   

       HC08_VC40 HC09_VC40 HC02_VC41 HC03_VC41 HC04_VC41 HC06_VC41 HC07_VC41  \
count      33179     33179     33179     33179     33179     33179     33179   
unique       347      1102       525       526       529         2         2   
top          (X)       (X)       0.0       0.0       0.0       (X)       (X)   
freq         946      1385       935       942       946     33178     33178   

       HC08_VC41 HC09_VC41  
count      33179     33179  
unique         2         2  
top          (X)       (X)  
freq       33178     33178  

[4 rows x 290 columns]

In [5]:
obama.describe()

GEO.id GEO.id2 GEO.display-label SUBHD0101_S01 SUBHD0102_S01  \
count            33121   33121             33121         33121         33121   
unique           33121   33121             33121         15428         11768   
top     8600000US21703   72683       ZCTA5 66526             0             0   
freq                 1       1                 1           144           152   

       SUBHD0103_S01 SUBHD0201_S01 SUBHD0202_S01 SUBHD0203_S01 HD03_S01  ...  \
count          33121         33121         33121         33121    33121  ...   
unique         11957             3             3             3     1358  ...   
top                0         100.0         100.0         100.0    100.0  ...   
freq             157         32976         32968         32963      431  ...   

       SUBHD0202_S40 SUBHD0203_S40 HD03_S40 SUBHD0101_S41 SUBHD0102_S41  \
count          33121         33121    33121         33121         33121   
unique           320           361     1189           547           540   
top              4.8           0.0    100.0          41.8          40.5   
freq             633           552     1000           286           331   

       SUBHD0103_S41 SUBHD0201_S41 SUBHD0202_S41 SUBHD0203_S41 HD03_S41  
count          33121         33121         33121         33121    33121  
unique           553             2             2             2        2  
top             43.5        ( X )         ( X )         ( X )    ( X )   
freq             328         33120         33120         33120    33120  

[4 rows x 290 columns]

In [6]:
bush.head()

GEO.id GEO.id2 GEO.display-label  \
0              Id     Id2         Geography   
1  8600000US00601   00601       ZCTA5 00601   
2  8600000US00602   00602       ZCTA5 00602   
3  8600000US00603   00603       ZCTA5 00603   
4  8600000US00604   00604       ZCTA5 00604   

                               HC02_VC01                        HC03_VC01  \
0  Number - Both sexes; Total population  Number - Male; Total population   
1                                  19143                             9369   
2                                  42042                            20608   
3                                  55530                            27271   
4                                   3923                             1911   

                           HC04_VC01                               HC06_VC01  \
0  Number - Female; Total population  Percent - Both sexes; Total population   
1                               9774                                   100.0   
2                              21434                                   100.0   
3                              28259                                   100.0   
4                               2012                                   100.0   

                          HC07_VC01                           HC08_VC01  \
0  Percent - Male; Total population  Percent - Female; Total population   
1                             100.0                               100.0   
2                             100.0                               100.0   
3                             100.0                               100.0   
4                             100.0                               100.0   

                                 HC09_VC01  ...  \
0  Males per 100 females; Total population  ...   
1                                     95.9  ...   
2                                     96.1  ...   
3                                     96.5  ...   
4                                     95.0  ...   

                                           HC07_VC40  \
0  Percent - Male; Total population - 75 years an...   
1                                                4.3   
2                                                3.4   
3                                                4.6   
4                                                1.9   

                                           HC08_VC40  \
0  Percent - Female; Total population - 75 years ...   
1                                                5.1   
2                                                3.7   
3                                                5.5   
4                                                1.7   

                                           HC09_VC40  \
0  Males per 100 females; Total population - 75 y...   
1                                               80.5   
2                                               87.1   
3                                               80.7   
4                                              108.8   

                                           HC02_VC41  \
0  Number - Both sexes; Total population - Median...   
1                                               30.3   
2                                               31.3   
3                                               33.4   
4                                               25.2   

                                           HC03_VC41  \
0  Number - Male; Total population - Median age (...   
1                                               28.9   
2                                               30.4   
3                                               31.9   
4                                               23.0   

                                           HC04_VC41  \
0  Number - Female; Total population - Median age...   
1                                               32.0   
2                                               32.0   
3                                               34.8   
4                                               27.3   

    

In [7]:
obama.head()

GEO.id GEO.id2 GEO.display-label  \
0              Id     Id2         Geography   
1  8600000US00601   00601       ZCTA5 00601   
2  8600000US00602   00602       ZCTA5 00602   
3  8600000US00603   00603       ZCTA5 00603   
4  8600000US00606   00606       ZCTA5 00606   

                           SUBHD0101_S01                    SUBHD0102_S01  \
0  Number - Both sexes; Total population  Number - Male; Total population   
1                                  18570                             9078   
2                                  41520                            20396   
3                                  54689                            26597   
4                                   6615                             3268   

                       SUBHD0103_S01                           SUBHD0201_S01  \
0  Number - Female; Total population  Percent - Both sexes; Total population   
1                               9492                                   100.0   
2                              21124                                   100.0   
3                              28092                                   100.0   
4                               3347                                   100.0   

                      SUBHD0202_S01                       SUBHD0203_S01  \
0  Percent - Male; Total population  Percent - Female; Total population   
1                             100.0                               100.0   
2                             100.0                               100.0   
3                             100.0                               100.0   
4                             100.0                               100.0   

                                  HD03_S01  ...  \
0  Males per 100 females; Total population  ...   
1                                     95.6  ...   
2                                     96.6  ...   
3                                     94.7  ...   
4                                     97.6  ...   

                                       SUBHD0202_S40  \
0  Percent - Male; Total population - 75 years an...   
1                                                5.0   
2                                                4.2   
3                                                5.5   
4                                                5.2   

                                       SUBHD0203_S40  \
0  Percent - Female; Total population - 75 years ...   
1                                                5.9   
2                                                5.2   
3                                                7.1   
4                                                5.3   

                                            HD03_S40  \
0  Males per 100 females; Total population - 75 y...   
1                                               81.4   
2                                               77.4   
3                                               73.4   
4                                               96.6   

                                       SUBHD0101_S41  \
0  Number - Both sexes; Total population - Median...   
1                                               35.9   
2                                               37.5   
3                                               38.2   
4                                               36.2   

                                       SUBHD0102_S41  \
0  Number - Male; Total population - Median age (...   
1                                               34.5   
2                                               36.6   
3                                               36.6   
4                                               35.7   

                                       SUBHD0103_S41  \
0  Number - Female; Total population - Median age...   
1                                               37.1   
2                                               38.5   
3                                               39.8   
4                                               36.6   

    

In [8]:
# compare dataframes

'''coded column names 
    >> e.g. SUBHD0103_S01, HD03_S41'''
# 2000 high-level column names (current exists before row 0 )
b_hi_column_names = [ name for name in bush.columns.values ]
# 2010 high-level column names (current exists before row 0 )
o_hi_column_names = [ name for name in obama.columns.values ]

# number of names are same 
if len( b_hi_column_names ) == len( o_hi_column_names ):
    # keep track of names that exist in both data
    hi_both = []   
    # keep track of 2000 names that do not exist in 2010 data
    hi_b_only = []
    # keep track of 2010 names that do not exist in 2000 data
    hi_o_only = []
    # index length
    for _ in range( len( b_hi_column_names ) ):
        # column name found at this index in 2000 df is not a column name in 2010 df 
        if b_hi_column_names[ _ ] not in o_hi_column_names:
            # add name to 2000 only list
            hi_b_only.append( b_hi_column_names[ _ ] )
        # column name found at this index in 2000 df is not a column name in 2010 df 
        if o_hi_column_names[ _ ] not in b_hi_column_names:
            # add name to 2000 only list
            hi_o_only.append( o_hi_column_names[ _ ] )
        else:
            hi_both.append([b_hi_column_names[ _ ],o_hi_column_names[ _ ]])
            
    print(f'hi\nboth = {len(hi_both)}\n2000 but not 2010 = {len(hi_b_only)}\n2010 but not 2000 = {len(hi_o_only)}')


'''commonly understandable column names
    >> e.g. Number - Female; Total population, Males per 100 females; Total population - Medi...'''
# 2000 translated column names (current row 0)
b_lo_column_names = [ name for name in bush.iloc[ 0 ] ]
# 2010 translated column names (current row 0)
o_lo_column_names = [ name for name in obama.iloc[ 0 ] ]
# number of names are same 
if len( b_lo_column_names ) == len( o_lo_column_names ):
    # keep track of names that exist in both data
    lo_both = []   
    # keep track of 2000 names that do not exist in 2010 data
    lo_b_only = []
    # keep track of 2010 names that do not exist in 2000 data
    lo_o_only = []
    # index length
    for _ in range( len( b_lo_column_names ) ):
        # column name found at this index in 2000 df is not a column name in 2010 df 
        if b_lo_column_names[ _ ] not in o_lo_column_names:
            # add name to 2000 only list
            lo_b_only.append( b_lo_column_names[ _ ] )
        # column name found at this index in 2000 df is not a column name in 2010 df 
        if o_lo_column_names[ _ ] not in b_lo_column_names:
            # add name to 2000 only list
            lo_o_only.append( o_lo_column_names[ _ ] )
        else:
            lo_both.append([b_lo_column_names[ _ ],o_lo_column_names[ _ ]])
            
    print(f'lo\nboth = {len(lo_both)}\n2000 but not 2010 = {len(lo_b_only)}\n2010 but not 2000 = {len(lo_o_only)}')


'''first row of numbered data 
    >> e.g. 18570, 100.0'''
# 2000 0th row of data (current row 1)
b_0th_real = [ name for name in bush.iloc[ 1 ] ]
# 2010 0th row of data (current row 1)
o_0th_real = [ name for name in obama.iloc[ 1 ] ]

# number of names are same 
if len( b_0th_real ) == len( o_0th_real ):
    # keep track of names that exist in both data
    both_0th = []   
    # keep track of 2000 names that do not exist in 2010 data
    b_only_0th = []
    # keep track of 2010 names that do not exist in 2000 data
    o_only_0th = []
    # index length
    for _ in range( len( b_0th_real ) ):
        # column name found at this index in 2000 df is not a column name in 2010 df 
        if b_0th_real[ _ ] not in o_0th_real:
            # add name to 2000 only list
            b_only_0th.append( b_0th_real[ _ ] )
        # column name found at this index in 2000 df is not a column name in 2010 df 
        if o_0th_real[ _ ] not in b_0th_real:
            # add name to 2000 only list
            o_only_0th.append( o_0th_real[ _ ] )
        else:
            both_0th.append([b_0th_real[ _ ],o_0th_real[ _ ]])
            
    print(f'data\nboth = {len(both_0th)}\n2000 but not 2010 = {len(b_only_0th)}\n2010 but not 2000 = {len(o_only_0th)}')

hi
both = 3
2000 but not 2010 = 287
2010 but not 2000 = 287
lo
both = 290
2000 but not 2010 = 0
2010 but not 2000 = 0
data
both = 80
2000 but not 2010 = 208
2010 but not 2000 = 210


- ***notes***:
    - while the high level labels don't seem to coexist 
        - any actual column name in 2000 is seen in 2010 and vise verse
    - under 1/3 
        - of values seen in the first row of data for either the d2000 or the 2010 datasets
            - are seen in that same row on the other dataset 
- ***actions***:
    - repeat the tests but check for correcponding index instead of coexistance 

In [9]:
# compare on index 

'''coded column names 
    >> e.g. SUBHD0103_S01, HD03_S41'''
# 2000 high-level column names (current exists before row 0 )
b_hi_column_names = [ name for name in bush.columns.values ]
# 2010 high-level column names (current exists before row 0 )
o_hi_column_names = [ name for name in obama.columns.values ]

# number of names are same 
if len( b_hi_column_names ) == len( o_hi_column_names ):
    # values that are same 
    same_hi = []   
    # values that are not the same
    not_the_hi = []
    # index length
    for _ in range( len( b_hi_column_names ) ):
        # object found at this index in 2000 is not same object at this index in 2010
        if b_hi_column_names[ _ ] != o_hi_column_names[ _ ]:
            # add name to 2000 only list
            not_the_hi.append( [ _ , b_hi_column_names[ _ ] , o_hi_column_names[ _ ] ] )
        else:
            same_hi.append( [ _ , b_hi_column_names[ _ ] , o_hi_column_names[ _ ] ] )
            
    print(f'hi level\nboth the same = {len(same_hi)}\nnot the same = {len(not_the_hi)}')
    

'''commonly understandable column names
    >> e.g. Number - Female; Total population, Males per 100 females; Total population - Medi...'''
# 2000 translated column names (current row 0)
b_lo_column_names = [ name for name in bush.iloc[ 0 ] ]
# 2010 translated column names (current row 0)
o_lo_column_names = [ name for name in obama.iloc[ 0 ] ]
# number of names are same 
if len( b_lo_column_names ) == len( o_lo_column_names ):
    # values that are same 
    same_lo = []   
    # values that are not the same
    not_the_lo = []
    # index length
    for _ in range( len( b_lo_column_names ) ):
        # object found at this index in 2000 is not same object at this index in 2010
        if b_lo_column_names[ _ ] != o_lo_column_names[ _ ]:
            # add name to 2000 only list
            not_the_lo.append( [ _ , b_lo_column_names[ _ ] , o_lo_column_names[ _ ] ] )
        else:
            same_lo.append( [ _ , b_lo_column_names[ _ ] , o_lo_column_names[ _ ] ] )
            
    print(f'common\nboth the same = {len(same_lo)}\nnot the same = {len(not_the_lo)}')

'''first row of numbered data 
    >> e.g. 18570, 100.0'''
# 2000 0th row of data (current row 1)
b_0th_real = [ name for name in bush.iloc[ 1 ] ]
# 2010 0th row of data (current row 1)
o_0th_real = [ name for name in obama.iloc[ 1 ] ]

# number of names are same 
if len( b_0th_real ) == len( o_0th_real ):
    # values that are same 
    same_0th = []   
    # values that are not the same
    not_the_0th = []
    # index length
    for _ in range( len( b_0th_real ) ):
        # object found at this index in 2000 is not same object at this index in 2010
        if o_0th_real[ _ ] != b_0th_real[ _ ]:
            # add name to 2000 only list
            not_the_0th.append( [ _ , b_0th_real[ _ ] , o_0th_real[ _ ] ] )
        else:
            same_0th.append( [ _ , b_0th_real[ _ ] , o_0th_real[ _ ] ] )
            
    print(f'data\nboth the same = {len(same_0th)}\nnot the same = {len(not_the_0th)}')

hi level
both the same = 3
not the same = 287
common
both the same = 290
not the same = 0
data
both the same = 13
not the same = 277


- ***notes***:
    - hi level
        - first 3 are the same
            - as expected
    - commonly understandable column names
        - perfect match
            - as expected
    - first row of data
        - less than 4.5% the same
            - about 85% fewer occourances than when testing existance of same instance in other df
        - good
- ***actions***:
    - set column names to commonly understandable column names
    - visualize change from 2000 to 2010

In [10]:
# copy dataframes
# 2000 Census
b = bush.copy()
# 2010 Census
o = obama.copy()

# reset 2000 columns to current 0th row values
b.columns = b.iloc[0]
# new 2000 dataframe without row where values are from
b = b[1:]

# reset 2010 columns to current 0th row values
o.columns = o.iloc[0]
# new 2010 dataframe without row where values are from
o = o[1:]

- ***actions***:
    - find commonalities to compare by
        - Id , Id2 , Geography
            - seem to be good possibilities

In [11]:
try_these = [ 'Id' , 'Id2' , 'Geography' ]

for column_name in try_these:
    print(column_name)
    same = []
    diff = []
    # len 2000 is 58 more than len 2010 
    for _ in range( 1 , len( b[ column_name ] )-58 ):
        if b[ column_name ][ _ ] == o[ column_name ][ _ ]:
            same.append( [_,b[ column_name ][ _ ],o[ column_name ][ _ ]] )
        else:
            diff.append( [_,b[ column_name ][ _ ],o[ column_name ][ _ ]] )
    print(f'same = {len(same)}\ndiff = {len(diff)}')
    i_same = [ instance.pop(0) for instance in same ] 
    # print(i_same)

Id
same = 165
diff = 32954
Id2
same = 165
diff = 32954
Geography
same = 165
diff = 32954


In [12]:
# 3 lists just output
by_id = [1, 2, 3, 44, 88, 89, 90, 19635, 19637, 19638, 19639, 19640, 19641, 19642, 19643, 19644, 19645, 19646, 19647, 19648, 19649, 19650, 19651, 19652, 19653, 19743, 19744, 20642, 20643, 20644, 20645, 20646, 20647, 20648, 20649, 20650, 20651, 20652, 20653, 20654, 20655, 20656, 20657, 20658, 20659, 20812, 20813, 20814, 20815, 20816, 20817, 20818, 20819, 20820, 20821, 20822, 20823, 20824, 20825, 20826, 20827, 20828, 20893, 20894, 20895, 20896, 20897, 20898, 20899, 20900, 20901, 20902, 20903, 20904, 20905, 20906, 20907, 20908, 20992, 20993, 20994, 20995, 20996, 20997, 20998, 20999, 21000, 21001, 21002, 21003, 21004, 21005, 21006, 21007, 21008, 21009, 21010, 21011, 21012, 21013, 21014, 21064, 21065, 21066, 21067, 21068, 21069, 21070, 21071, 21072, 21073, 21074, 21075, 21076, 21077, 21078, 21079, 21080, 21081, 21082, 21083, 21084, 21085, 21086, 21087, 21088, 21089, 21090, 21091, 21092, 21093, 21094, 21523, 21524, 21778, 21779, 21780, 21781, 21782, 21783, 21784, 21785, 21786, 21787, 21788, 21789, 21790, 21791, 21792, 21793, 21794, 21839, 21840, 22138, 22146, 22147, 22148, 22149, 22150, 22151, 22152, 22225, 22226, 22227, 22228]
by_id2 = [1, 2, 3, 44, 88, 89, 90, 19635, 19637, 19638, 19639, 19640, 19641, 19642, 19643, 19644, 19645, 19646, 19647, 19648, 19649, 19650, 19651, 19652, 19653, 19743, 19744, 20642, 20643, 20644, 20645, 20646, 20647, 20648, 20649, 20650, 20651, 20652, 20653, 20654, 20655, 20656, 20657, 20658, 20659, 20812, 20813, 20814, 20815, 20816, 20817, 20818, 20819, 20820, 20821, 20822, 20823, 20824, 20825, 20826, 20827, 20828, 20893, 20894, 20895, 20896, 20897, 20898, 20899, 20900, 20901, 20902, 20903, 20904, 20905, 20906, 20907, 20908, 20992, 20993, 20994, 20995, 20996, 20997, 20998, 20999, 21000, 21001, 21002, 21003, 21004, 21005, 21006, 21007, 21008, 21009, 21010, 21011, 21012, 21013, 21014, 21064, 21065, 21066, 21067, 21068, 21069, 21070, 21071, 21072, 21073, 21074, 21075, 21076, 21077, 21078, 21079, 21080, 21081, 21082, 21083, 21084, 21085, 21086, 21087, 21088, 21089, 21090, 21091, 21092, 21093, 21094, 21523, 21524, 21778, 21779, 21780, 21781, 21782, 21783, 21784, 21785, 21786, 21787, 21788, 21789, 21790, 21791, 21792, 21793, 21794, 21839, 21840, 22138, 22146, 22147, 22148, 22149, 22150, 22151, 22152, 22225, 22226, 22227, 22228]
by_geo = [1, 2, 3, 44, 88, 89, 90, 19635, 19637, 19638, 19639, 19640, 19641, 19642, 19643, 19644, 19645, 19646, 19647, 19648, 19649, 19650, 19651, 19652, 19653, 19743, 19744, 20642, 20643, 20644, 20645, 20646, 20647, 20648, 20649, 20650, 20651, 20652, 20653, 20654, 20655, 20656, 20657, 20658, 20659, 20812, 20813, 20814, 20815, 20816, 20817, 20818, 20819, 20820, 20821, 20822, 20823, 20824, 20825, 20826, 20827, 20828, 20893, 20894, 20895, 20896, 20897, 20898, 20899, 20900, 20901, 20902, 20903, 20904, 20905, 20906, 20907, 20908, 20992, 20993, 20994, 20995, 20996, 20997, 20998, 20999, 21000, 21001, 21002, 21003, 21004, 21005, 21006, 21007, 21008, 21009, 21010, 21011, 21012, 21013, 21014, 21064, 21065, 21066, 21067, 21068, 21069, 21070, 21071, 21072, 21073, 21074, 21075, 21076, 21077, 21078, 21079, 21080, 21081, 21082, 21083, 21084, 21085, 21086, 21087, 21088, 21089, 21090, 21091, 21092, 21093, 21094, 21523, 21524, 21778, 21779, 21780, 21781, 21782, 21783, 21784, 21785, 21786, 21787, 21788, 21789, 21790, 21791, 21792, 21793, 21794, 21839, 21840, 22138, 22146, 22147, 22148, 22149, 22150, 22151, 22152, 22225, 22226, 22227, 22228]

# these lists are the same
if not by_id == by_id2 == by_geo:
    raise Exception('FALSE == by_id == by_id2 == by_geo == False')

# and for those 3 columns* every index value in any list is the same in 2000 and 2010 *(Id, Id2, Geography)
for i in [ by_id , by_id2 , by_geo ]:
    for _ in i:
        # compare Id values
        if b.Id[_] != o.Id[_]:
            # if any don't match 2000 vs 2010
            raise Exception(f'b.Id[_] != o.Id[_]\n{b.Id[_]} != {o.Id[_]}')
        # compare Id2 values
        if b.Id2[_] != o.Id2[_]:
            # if any don't match 2000 vs 2010
            raise Exception(f'b.Id2[_] != o.Id2[_]\n{b.Id2[_]} != {o.Id2[_]}')
        # compare Geography values
        if b.Geography[_] != o.Geography[_]:
            # if any don't match 2000 vs 2010
            raise Exception(f'b.Geography[_] != o.Geography[_]\n{b.Geography[_]} != {o.Geography[_]}')


- ***notes***:
    - why are these values the same?
    - what do they mean/represent?
- ***actions***:
    - answer above

In [13]:
len(o.Geography)

33120

In [14]:
last_5_2000_geo = [ b.Geography[_][-5:] for _ in range( 1 , len(b.Geography)+1) ]
last_5_2010_geo = [ o.Geography[_][-5:] for _ in range( 1 , len(o.Geography)+1) ]

# note the index patterns
if b.Geography[33113][-5:] == last_5_2000_geo[33112] == o.Geography[33062][-5:] == last_5_2010_geo[33061]:
    pass
    if b.Geography[33114][-5:] == last_5_2000_geo[33113] == o.Geography[33063][-5:] == last_5_2010_geo[33062]:
        pass
        if b.Geography[33110][-5:] == last_5_2000_geo[33109] == o.Geography[33059][-5:] == last_5_2010_geo[33058]:
            pass
        else:
            raise Exception('ERROR2')
    else:
        raise Exception('ERROR1')
else:
    raise Exception('ERROR')

In [15]:
# each 5 digit code from end of each row in 2010 'Geography'
bgeo = [ _[-5:] for _ in b.Geography ]

# number of coexisting instances
x=0
# collect coexisting instances
collect_locs = []
# collect unique coexisting instances
uni_col_locs = set() 
# len 2010 is -58 vs len 2000 
for _ in range(1,len(o.Geography)):
    # 5 digit code is in 2010 collection 
    if o.Geography[_][-5:] in bgeo:
        # tag it
        x += 1
        # bag it
        collect_locs.append(o.Geography[_][-5:])
        uni_col_locs.add(o.Geography[_][-5:])


# number of coexisting , size of that list , number of unique , len 2000 vs 2010
x , len(collect_locs) , len(list(uni_col_locs)) , len(b.Geography) - len(o.Geography)

(31377, 31377, 31377, 58)

- ***notes***:
    - ok, this is large
        - readings credit
            - did a few pages of this https://www.census.gov/prod/cen2000/doc/sf1.pdf 
                - 494-497 + skim 
            - followed by https://www.quora.com/U-S-Census-Bureau-How-does-Zip-Code-Tabulation-Area-is-created 
    - in short,
        - the US Census Bureau doesn't strictly use postal zip code when determining 5-digit zip
            - leave out some very small zips
            - made zips for large bodies of water and land which did not have zip codes
                - this was done in 2000 and not done in 2010
    - what we've found:
        - all 5 digit codes (found at end of each object) in 2010 'Geography' column 
            - are seen at the end of an object in the 2000 'Geograpy' column 
- ***actions***:
    - drop columns where 5-digit is not seen
        - see if df is then ready for straight compare

In [61]:
# identify zip codes from 2000 .Geography (last 5 chars of string)
zip_2000_codes = [q[-5:] for q in b.Geography]  # ValueError: invalid literal for int() with base 10: '006HH'
# identify zip codes from 2010 .Geography (last 5 chars of string)
zip_2010_codes = [q[-5:] for q in o.Geography]

# enumerate 2000
e_zip_y2k = enumerate(zip_2000_codes)
# enumerate 2010
e_zip_2k10 = enumerate(zip_2010_codes)

In [62]:
# from 2000.Geography , instance is not seen in 2010.Geography
a = [zip_code for zip_code in zip_2000_codes if zip_code not in zip_2010_codes]
# from 2010.Geography , instance is not seen in 2000.Geography
b = [zip_code for zip_code in zip_2010_codes if zip_code not in zip_2000_codes]
# from 2000.Geography , instance is seen in 2010.Geography
c = [zip_code for zip_code in zip_2000_codes if zip_code in zip_2010_codes]
# from 2010.Geography , instance is seen in 2000.Geography
d = [zip_code for zip_code in zip_2010_codes if zip_code in zip_2000_codes]

In [63]:
# from 2000.Geography , instance is not seen in 2010.Geography  -- sample: zip_code = (2, 'c')
aa = [zip_code for zip_code in e_zip_y2k if zip_code[1] not in zip_2010_codes]
# from 2010.Geography , instance is not seen in 2000.Geography  -- sample: zip_code[1] = 'c'
bb = [zip_code for zip_code in e_zip_2k10 if zip_code[1] not in zip_2000_codes]

# from 2000.Geography , instance is seen in 2010.Geography
cc = [zip_code for zip_code in e_zip_y2k if zip_code[1] in zip_2010_codes]
# from 2010.Geography , instance is seen in 2000.Geography
dd = [zip_code for zip_code in e_zip_2k10 if zip_code[1] in zip_2000_codes]

In [64]:
len(a),len(aa),len(b),len(bb),len(c),len(cc),len(d),len(dd)

(1800, 1800, 1743, 1743, 31377, 0, 31377, 0)

In [18]:
# print(f'len(in_2000_but_not_2010_from_2000) = {len(in_2000_but_not_2010_from_2000)}\nlen(in_2010_but_not_2000_from_2010) = {len(in_2010_but_not_2000_from_2010)}\nlen(in_2000_and_2010_from_2000) = {len(in_2000_and_2010_from_2000)}\nlen(in_2010_and_2000_from_2010) = {len(in_2010_and_2000_from_2010)}')

len(in_2000_but_not_2010_from_2000) = 1800
len(in_2010_but_not_2000_from_2010) = 1742
len(in_2000_and_2010_from_2000) = 31378
len(in_2010_and_2000_from_2010) = 31378


- ***notes***:
    - len(in_2000_and_2010_from_2000) == len(in_2010_and_2000_from_2010) == 31378
        - big relief 
- ***actions***:
    - convert 2010 and 2000 dataframes based on shared values in .Geography 
        - and
            - drop 'ZCTA5' form 'Geography' rows
            - check that in_2000_but_not_2010_from_2000 doesn't share with in_2010_but_not_2000_from_2010
        - potential issues
            - same 5-digit zip code is not actually same place
        - solutions
            - any changes of this sort have been documented
                - review prior readings on overview
    - below before editing df is confirming assumptions and provinding evidence for df editing strats

In [30]:
for _ in in_2000_but_not_2010_from_2000:
    if _ in in_2010_but_not_2000_from_2010:
        raise Exception(f'{_} in in_2010_but_not_2000_from_2010')
for _ in in_2010_but_not_2000_from_2010:
    if _ in in_2000_but_not_2010_from_2000:
        raise Exception(f'{_} in in_2000_but_not_2010_from_2000')

In [31]:
lettered = []
no_letters = []
for _ in in_2000_but_not_2010_from_2000:
    if isinstance(_,str):
        if 'X' in _ or 'H' in _:
            lettered.append(_)
        elif 'X' not in _ and 'H' not in _:
            no_letters.append(int(_))
        else:
            raise Exception(_)
    else:
        raise Exception(f'NON STR INSTANCE , {type(_)}')

for _ in lettered:
    if _ in in_2010_but_not_2000_from_2010:
        raise Exception(_)

for _ in no_letters:
    if _ in in_2010_but_not_2000_from_2010:
        raise Exception(_)

In [38]:
print(f'lettered = {len(lettered)}\nno_letters = {len(no_letters)}')

lettered = 1140
no_letters = 660


In [39]:
lettered2 = []
no_letters2 = []
for _ in in_2010_but_not_2000_from_2010:
    if isinstance(_,str):
        if 'X' in _ or 'H' in _:
            lettered2.append(_)
        elif 'X' not in _ and 'H' not in _:
            no_letters2.append(int(_))
        else:
            raise Exception(_)
    else:
        raise Exception(f'NON STR INSTANCE , {type(_)}')

In [41]:
print(f'lettered = {len(lettered)}\nlettered2 = {len(lettered2)}\nno_letters = {len(no_letters)}\nno_letters2 = {len(no_letters2)}')

lettered = 1140
lettered2 = 0
no_letters = 660
no_letters2 = 1742


In [42]:
"""in_2010_but_not_2000_from_2010 and in_2000_but_not_2010_from_2000 are completely different""" 

# 2010_not_2000 vs 2000_not_2010
for _ in in_2010_but_not_2000_from_2010:
    if _ in in_2000_but_not_2010_from_2000:
        raise Exception(f'in_2010_but_not_2000_from_2010 {_}')

# 2000_not_2010 vs 2010_not_2000
for _ in in_2000_but_not_2010_from_2000:
    if _ in in_2010_but_not_2000_from_2010:
        raise Exception(f'in_2010_but_not_2000_from_2010 {_}')

In [43]:
for _ in range(len(in_2010_and_2000_from_2010)):
    if in_2010_and_2000_from_2010[_] != in_2000_and_2010_from_2000[_]:
        raise Exception('!= ERROR != ERROR !=')

In [44]:
# set copy
__b__ = b.copy()

# label objects in 2000.Geography with their index
nume_b_geo = enumerate(b.Geography)
# label objects in 2010.Geography with their index
nume_b_geo = enumerate(o.Geography)

# identify codes occouring in 2000 and not 2010
_2000_not_2010 = [i for (i,_) in nume_b_geo if _[-5:] not in zip_2010_codes]

# index and instance for all 2000 Census 'Geography' rows 
# if the zip code (last 5 digits) is NOT in the collection of 2010 Census zip codes 
# drop row index from the 2000 Census 
# ready_b = b.copy().Geography.drop([i for (i,_) in nume_b_geo if _[-5:] not in zip_2010_codes])
ready_b = b.copy().drop()

of_indexes = [i for i,j in in_2000_and_2010_from_2000]
check_indexes = [i for i,j in in_2010_and_2000_from_2010]
if of_indexes == check_indexes:
    for _ in range(len(in_2000_and_2010_from_2000)):
        if _ not in of_indexes:
            __b__.drop([_])

00601
00602
00603
00606
00610
00612
00616
00617
00622
00623
00624
00627
00631
00637
00638
00641
00646
00647
00650
00652
00653
00656
00659
00660
00662
00664
00667
00669
00670
00674
00676
00677
00678
00680
00682
00683
00685
00687
00688
00690
00692
00693
00698
00703
00704
00705
00707
00714
00715
00716
00717
00718
00719
00720
00723
00725
00728
00729
00730
00731
00735
00736
00738
00739
00740
00741
00745
00751
00754
00757
00765
00766
00767
00769
00771
00772
00773
00775
00777
00778
00780
00782
00783
00784
00791
00794
00795
00901
00906
00907
00909
00911
00912
00913
00915
00917
00918
00920
00921
00923
00924
00925
00926
00927
00934
00949
00952
00953
00956
00957
00959
00961
00962
00965
00966
00968
00969
00971
00976
00979
00982
00983
00985
00987
01001
01002
01005
01007
01008
01010
01011
01012
01013
01020
01022
01026
01027
01028
01029
01030
01031
01032
01033
01034
01035
01036
01037
01038
01039
01040
01050
01053
01054
01056
01057
01060
01062
01068
01069
01070
01071
01072
01073
01075
01077
01080
0108

08217
08221
08223
08224
08225
08226
08230
08232
08234
08241
08242
08243
08244
08247
08248
08251
08260
08270
08302
08310
08311
08312
08314
08316
08317
08318
08319
08321
08322
08323
08324
08326
08327
08328
08329
08330
08332
08340
08341
08343
08344
08345
08346
08348
08349
08350
08352
08353
08360
08361
08401
08402
08403
08406
08501
08502
08505
08510
08511
08512
08514
08515
08518
08520
08525
08527
08528
08530
08533
08534
08535
08536
08540
08542
08550
08551
08553
08554
08555
08558
08559
08560
08561
08562
08608
08609
08610
08611
08618
08619
08620
08628
08629
08638
08640
08641
08648
08690
08691
08701
08720
08721
08722
08723
08724
08730
08731
08732
08733
08734
08735
08736
08738
08740
08741
08742
08750
08751
08752
08753
08755
08757
08758
08759
08801
08802
08804
08805
08807
08809
08810
08812
08816
08817
08820
08821
08822
08823
08824
08825
08826
08827
08828
08829
08830
08831
08832
08833
08835
08836
08837
08840
08846
08848
08850
08852
08853
08854
08857
08858
08859
08861
08863
08865
08867
08869
0887

15650
15655
15656
15658
15660
15661
15662
15663
15665
15666
15668
15670
15671
15672
15673
15675
15676
15677
15678
15679
15680
15681
15683
15684
15686
15687
15688
15689
15690
15692
15693
15695
15696
15697
15698
15701
15710
15711
15712
15713
15714
15715
15716
15717
15721
15722
15723
15724
15725
15728
15729
15730
15731
15732
15733
15734
15736
15739
15741
15742
15744
15745
15747
15748
15752
15753
15754
15756
15757
15759
15760
15761
15762
15764
15765
15767
15770
15771
15772
15773
15774
15775
15776
15777
15778
15779
15780
15781
15783
15784
15801
15821
15823
15824
15825
15827
15828
15829
15832
15834
15840
15841
15845
15846
15847
15848
15849
15851
15853
15856
15857
15860
15861
15863
15864
15865
15866
15868
15870
15901
15902
15904
15905
15906
15909
15920
15923
15924
15925
15926
15927
15928
15929
15930
15931
15934
15935
15936
15937
15938
15940
15942
15943
15944
15946
15948
15949
15952
15954
15955
15956
15957
15958
15961
15963
16001
16002
16020
16022
16023
16024
16025
16027
16028
16029
16030
1603

22728
22729
22730
22731
22732
22733
22734
22735
22736
22737
22738
22740
22741
22742
22743
22747
22749
22801
22802
22810
22811
22812
22815
22820
22821
22824
22827
22830
22831
22832
22834
22835
22840
22841
22842
22843
22844
22845
22846
22847
22849
22850
22851
22853
22901
22902
22903
22911
22920
22922
22923
22931
22932
22935
22936
22937
22938
22939
22940
22942
22943
22946
22947
22948
22949
22952
22958
22959
22960
22963
22964
22967
22968
22969
22971
22972
22973
22974
22976
22980
22989
23002
23004
23005
23009
23011
23015
23021
23022
23023
23024
23025
23027
23030
23032
23035
23038
23039
23040
23043
23045
23047
23050
23055
23056
23059
23060
23061
23062
23063
23064
23065
23066
23068
23069
23070
23071
23072
23075
23076
23079
23083
23084
23085
23086
23089
23091
23092
23093
23102
23103
23106
23108
23109
23110
23111
23112
23113
23115
23116
23117
23119
23120
23123
23124
23125
23126
23128
23129
23130
23138
23139
23140
23141
23146
23148
23149
23150
23153
23156
23161
23163
23168
23169
23175
23176
2317

29212
29223
29229
29301
29302
29303
29306
29307
29316
29320
29321
29322
29323
29325
29330
29332
29334
29335
29338
29340
29341
29349
29351
29353
29355
29356
29360
29364
29365
29369
29370
29372
29373
29374
29375
29376
29377
29379
29384
29385
29388
29401
29403
29404
29405
29406
29407
29412
29414
29418
29420
29423
29426
29429
29431
29432
29434
29435
29436
29437
29438
29439
29440
29445
29446
29448
29449
29450
29451
29452
29453
29455
29456
29458
29461
29464
29466
29468
29469
29470
29471
29472
29474
29475
29477
29479
29481
29482
29483
29485
29487
29488
29492
29493
29501
29505
29506
29510
29511
29512
29516
29518
29519
29520
29525
29526
29527
29530
29532
29536
29540
29541
29543
29544
29545
29546
29547
29550
29554
29555
29556
29560
29563
29564
29565
29566
29567
29568
29569
29570
29571
29572
29574
29575
29576
29577
29579
29580
29581
29582
29583
29584
29585
29590
29591
29592
29593
29594
29596
29601
29605
29607
29609
29611
29615
29617
29620
29621
29624
29625
29626
29627
29628
29630
29631
29635
2963

36264
36265
36266
36267
36268
36269
36271
36272
36273
36274
36276
36277
36278
36279
36280
36301
36303
36305
36310
36311
36312
36313
36314
36316
36317
36318
36319
36320
36322
36323
36330
36340
36343
36344
36345
36346
36350
36351
36352
36353
36360
36362
36370
36371
36373
36374
36375
36376
36401
36420
36425
36426
36432
36435
36436
36441
36442
36444
36445
36446
36451
36453
36455
36456
36460
36467
36470
36471
36473
36474
36475
36476
36477
36480
36481
36482
36483
36502
36505
36507
36509
36511
36513
36518
36521
36522
36523
36524
36525
36526
36527
36528
36529
36530
36532
36535
36538
36539
36540
36541
36542
36544
36545
36548
36549
36550
36551
36553
36556
36558
36559
36560
36561
36562
36564
36567
36569
36571
36572
36574
36575
36576
36578
36579
36580
36581
36582
36583
36584
36585
36587
36602
36603
36604
36605
36606
36607
36608
36609
36610
36611
36612
36613
36617
36618
36619
36693
36695
36701
36703
36720
36722
36723
36726
36727
36728
36732
36736
36738
36740
36742
36744
36748
36749
36750
36751
3675

43620
43623
43701
43713
43716
43717
43718
43719
43720
43721
43722
43723
43724
43725
43727
43728
43730
43731
43732
43734
43735
43738
43739
43740
43746
43747
43748
43749
43754
43755
43756
43758
43760
43762
43764
43766
43767
43768
43771
43772
43773
43777
43778
43779
43780
43782
43783
43787
43788
43793
43802
43804
43811
43812
43821
43822
43824
43830
43832
43837
43840
43842
43843
43844
43845
43901
43902
43903
43905
43906
43907
43908
43910
43912
43913
43914
43915
43917
43920
43925
43926
43927
43928
43930
43932
43933
43934
43935
43938
43939
43940
43942
43943
43944
43945
43946
43947
43948
43950
43951
43952
43953
43961
43962
43963
43964
43967
43968
43971
43972
43973
43974
43976
43977
43983
43986
43988
44001
44003
44004
44010
44011
44012
44017
44021
44022
44023
44024
44026
44028
44030
44032
44035
44039
44040
44041
44044
44046
44047
44048
44049
44050
44052
44053
44054
44055
44056
44057
44060
44062
44064
44065
44067
44070
44072
44074
44076
44077
44080
44081
44082
44084
44085
44086
44087
44089
4409

49339
49340
49341
49342
49343
49344
49345
49346
49347
49348
49349
49401
49402
49403
49404
49405
49406
49408
49410
49411
49412
49415
49417
49418
49419
49420
49421
49423
49424
49425
49426
49428
49431
49434
49435
49436
49437
49440
49441
49442
49444
49445
49446
49448
49449
49450
49451
49452
49453
49454
49455
49456
49457
49458
49459
49460
49461
49464
49503
49504
49505
49506
49507
49508
49509
49512
49525
49544
49546
49548
49601
49611
49612
49613
49614
49615
49616
49617
49618
49619
49620
49621
49622
49623
49625
49626
49627
49628
49629
49630
49631
49632
49633
49634
49635
49636
49637
49638
49639
49640
49642
49643
49644
49645
49646
49648
49649
49650
49651
49653
49654
49655
49656
49657
49659
49660
49663
49664
49665
49666
49667
49668
49670
49675
49676
49677
49679
49680
49682
49683
49684
49686
49688
49689
49690
49701
49705
49706
49707
49709
49710
49712
49713
49715
49716
49718
49719
49720
49721
49722
49724
49725
49726
49727
49728
49729
49730
49733
49735
49736
49738
49740
49743
49744
49745
49746
4974

55970
55971
55972
55973
55974
55975
55976
55977
55979
55981
55982
55983
55985
55987
55990
55991
55992
56001
56003
56007
56009
56010
56011
56013
56014
56016
56017
56019
56020
56021
56022
56023
56024
56025
56026
56027
56028
56029
56031
56032
56033
56034
56035
56036
56037
56039
56041
56042
56043
56044
56045
56048
56050
56051
56052
56054
56055
56056
56057
56058
56060
56062
56063
56065
56068
56069
56071
56072
56073
56074
56075
56078
56080
56081
56082
56083
56085
56087
56088
56089
56090
56091
56093
56096
56097
56098
56101
56110
56111
56113
56114
56115
56116
56117
56118
56119
56120
56121
56122
56123
56127
56128
56129
56131
56132
56134
56136
56137
56138
56139
56140
56141
56142
56143
56144
56145
56147
56149
56150
56151
56152
56153
56155
56156
56157
56158
56159
56160
56161
56162
56164
56165
56166
56167
56168
56169
56170
56171
56172
56173
56174
56175
56176
56177
56178
56180
56181
56183
56185
56186
56187
56201
56207
56208
56209
56211
56212
56214
56215
56216
56218
56219
56220
56221
56222
56223
5622

61725
61726
61727
61728
61729
61730
61731
61732
61733
61734
61735
61736
61737
61738
61739
61740
61741
61742
61743
61744
61745
61747
61748
61749
61750
61752
61753
61754
61755
61756
61759
61760
61761
61764
61769
61770
61771
61772
61773
61774
61775
61776
61777
61778
61801
61802
61810
61811
61812
61813
61814
61815
61816
61817
61818
61820
61821
61822
61830
61831
61832
61833
61834
61839
61840
61841
61842
61843
61844
61845
61846
61847
61848
61849
61850
61851
61852
61853
61854
61855
61856
61857
61858
61859
61862
61863
61864
61865
61866
61870
61871
61872
61873
61874
61875
61876
61877
61878
61880
61882
61883
61884
61910
61911
61912
61913
61914
61917
61919
61920
61924
61925
61928
61929
61930
61931
61932
61933
61936
61937
61938
61940
61941
61942
61943
61944
61949
61951
61953
61956
61957
62001
62002
62006
62009
62010
62011
62012
62013
62014
62015
62016
62017
62018
62019
62021
62022
62023
62024
62025
62027
62028
62030
62031
62032
62033
62034
62035
62036
62037
62040
62044
62045
62046
62047
62048
6204

67648
67649
67650
67651
67653
67654
67656
67657
67658
67659
67660
67661
67663
67664
67665
67667
67669
67671
67672
67673
67674
67675
67701
67730
67731
67732
67733
67734
67735
67736
67737
67738
67739
67740
67741
67743
67744
67745
67747
67748
67749
67751
67752
67753
67756
67757
67758
67761
67762
67764
67801
67831
67834
67835
67836
67837
67838
67839
67840
67841
67842
67844
67846
67849
67850
67851
67853
67854
67855
67857
67859
67860
67861
67862
67863
67864
67865
67867
67868
67869
67870
67871
67876
67877
67878
67879
67880
67882
67901
67950
67951
67952
67953
67954
68001
68002
68003
68004
68005
68007
68008
68010
68014
68015
68016
68017
68018
68019
68020
68022
68023
68025
68028
68029
68030
68031
68033
68034
68036
68037
68038
68039
68040
68041
68042
68044
68045
68046
68047
68048
68050
68055
68057
68058
68059
68061
68062
68063
68064
68065
68066
68067
68068
68069
68070
68071
68073
68102
68104
68105
68106
68107
68108
68110
68111
68112
68113
68114
68116
68117
68118
68122
68123
68124
68127
68128
6813

74553
74554
74555
74556
74557
74558
74560
74561
74562
74563
74565
74567
74569
74570
74571
74572
74574
74576
74577
74578
74601
74604
74630
74631
74632
74633
74636
74637
74640
74641
74643
74644
74646
74647
74650
74651
74652
74653
74701
74720
74722
74723
74724
74726
74727
74728
74729
74730
74731
74733
74734
74735
74736
74738
74740
74741
74743
74745
74747
74748
74750
74753
74754
74755
74756
74759
74760
74761
74764
74766
74801
74804
74820
74824
74825
74826
74827
74829
74830
74831
74832
74833
74834
74836
74837
74839
74840
74842
74843
74844
74845
74848
74849
74850
74851
74852
74854
74855
74856
74857
74859
74860
74864
74865
74867
74868
74869
74871
74872
74873
74875
74878
74880
74881
74883
74884
74901
74902
74930
74931
74932
74935
74936
74937
74939
74940
74941
74942
74943
74944
74945
74946
74948
74949
74951
74953
74954
74955
74956
74957
74959
74960
74962
74963
74964
74965
74966
75001
75002
75006
75007
75009
75010
75013
75019
75020
75021
75022
75023
75024
75025
75028
75032
75034
75035
75038
7503

80863
80864
80903
80904
80905
80906
80907
80908
80909
80910
80911
80913
80915
80916
80917
80918
80919
80920
80921
80922
80925
80926
80928
80929
80930
81001
81003
81004
81005
81006
81007
81008
81020
81021
81022
81023
81024
81025
81027
81029
81030
81033
81036
81039
81040
81041
81043
81044
81045
81047
81049
81050
81052
81054
81055
81057
81058
81059
81062
81063
81064
81067
81069
81071
81073
81076
81077
81081
81082
81084
81087
81089
81090
81091
81092
81101
81120
81121
81122
81123
81124
81125
81126
81128
81130
81131
81132
81133
81136
81137
81140
81141
81143
81144
81146
81147
81148
81149
81151
81152
81154
81155
81201
81210
81211
81212
81220
81221
81222
81223
81224
81225
81226
81230
81233
81235
81236
81239
81240
81241
81243
81244
81248
81251
81252
81253
81301
81320
81321
81323
81324
81325
81326
81327
81328
81330
81331
81332
81334
81335
81401
81410
81411
81413
81415
81416
81418
81419
81422
81423
81424
81425
81426
81427
81428
81430
81431
81432
81433
81434
81435
81501
81503
81504
81505
81506
8152

92676
92677
92679
92683
92688
92691
92692
92694
92701
92703
92704
92705
92706
92707
92708
92780
92782
92801
92802
92804
92805
92806
92807
92808
92821
92823
92831
92832
92833
92835
92840
92841
92843
92844
92845
92860
92861
92865
92866
92867
92868
92869
92870
92879
92880
92881
92882
92883
92886
92887
93001
93003
93004
93010
93012
93013
93015
93021
93022
93023
93030
93033
93035
93040
93041
93060
93063
93065
93066
93067
93101
93103
93105
93108
93109
93110
93111
93117
93201
93202
93203
93204
93205
93206
93207
93208
93210
93212
93215
93218
93219
93221
93222
93223
93224
93225
93226
93230
93234
93235
93238
93239
93240
93241
93242
93243
93244
93245
93247
93249
93250
93251
93252
93254
93255
93256
93257
93260
93261
93262
93263
93265
93266
93267
93268
93270
93271
93272
93274
93276
93277
93280
93283
93285
93286
93287
93291
93292
93301
93304
93305
93306
93307
93308
93309
93311
93312
93313
93401
93402
93405
93420
93422
93424
93426
93427
93428
93429
93430
93432
93433
93434
93436
93437
93440
93441
9344

99755
99756
99757
99758
99759
99760
99761
99762
99763
99764
99765
99766
99767
99768
99769
99770
99771
99772
99773
99774
99775
99776
99777
99778
99780
99781
99782
99783
99784
99785
99786
99788
99789
99791
99801
99820
99824
99825
99826
99827
99829
99830
99832
99833
99835
99840
99841
99901
99903
99918
99919
99921
99922
99923
99925
99926
99927
99929


In [25]:
'''split'''
# # label objects in 2000.Geography with their index
# nume_b_geo = enumerate(b.Geography)
# # label objects in 2010.Geography with their index
# nume_b_geo = enumerate(o.Geography)

# # identify codes occouring in 2000 and not 2010
# [i for (i,_) in nume_b_geo if _[-5:] not in zip_2010_codes]

# # index and instance for all 2000 Census 'Geography' rows 
# # if the zip code (last 5 digits) is NOT in the collection of 2010 Census zip codes 
# # drop row index from the 2000 Census 
# # ready_b = b.copy().Geography.drop([i for (i,_) in nume_b_geo if _[-5:] not in zip_2010_codes])
# ready_b = b.copy().drop()


In [58]:
for i in enumerate(['a','b','c','d','e','f']):
    print(i)

(0, 'a')
(1, 'b')
(2, 'c')
(3, 'd')
(4, 'e')
(5, 'f')
